lib/rubocop/cop/style/file_name.rb

```diff
@@ -16,6 +16,9 @@ def investigate(processed_source)
           basename = File.basename(file_path).sub(/\.[^\.]+$/, '')
           return if snake_case?(basename)
 
+          first_line = processed_source.lines.first
```

Comment:
The change in behaviour should likely be reflected in the class comment as well.